source/tutorial/autocomplete-component.md

```diff
@@ -0,0 +1,164 @@
+As they search for a rental, a user might also want to narrow their search to a specific city. Let's build a component that will let them search for properties within a city, and also suggest cities to them as they type.
+
+To begin, let's generate our new component. We'll call this component `filter-listing`.
+
+```shell
+ember g component filter-listing
+```
+As before, this creates a Handlebars template (`app\templates\components\filter-listing.hbs`) and a JavaScript file (`app\components\filter-listing.js`).
+
+The Handlebars template looks like this:
+
+```app\templates\components\filter-listing.hbs
+City: {{input value=filter key-up=(action 'keyUp' filter)}} 
+<button {{action 'click'}}>Search</button>
+
+<ul>
+{{#each filteredList as |item|}}
+  <li {{action 'choose' item.city}}>{{item.city}}</li>
+{{/each}}
+</ul>
+```
+It contains an `input` component, that renders as a text field that the user can type in to look for properties in a given city. The `value` property of the `input` will be bound to the `filter` property in our backing JavaScript object. The `key-up` property will be bound to a `keyUp` action in our backing object, and passes the `filter` property as a parameter.
+
+It also contains a button, whose `action` parameter is bound to the `click` action in our component.
+
+Lastly, it contains an unordered list, that uses the `filteredList` property for data, and displays the `city` property of each item in the list. Clicking the list item will fire the `choose` action, which will populate the `input` field with the name of the `city` in the clicked list item.
+
+Here is what the component's JavaScript looks like:
+
+```app\components\filter-listing.js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  filter: null,
+  filteredList: null,
+  actions: {
+    keyUp() {
+      this.attrs.autoComplete(this.filter);
+    },
+    click() {
+      this.attrs.search(this.filter);
+    },
+    choose(city) {
+      this.set('filter',city);
+    }
+  }
+});
+
+```
+There's a property for each of the `filter` and `filteredList`, and actions as described above. What's interesting is that only the `choose` action is defined by the component. The actual logic of each of the `keyUp` and `click` actions are pulled from the component's `attrs` object, which means that those actions need to be passed in by the calling object, a pattern known as _closure actions_.
```

Comment:
Can you review the newline policy in the guidelines to make diffs/comments easier, please?

We shouldn't refer at all to `attrs`, check the wording on the relevant guide and you can even link there.